src/components/accessories/patientSummary/patientSummaryByDate/PatientSummaryByDate.tsx

```diff
@@ -1,43 +1,72 @@
-  { date: "21/12/2019", type: "Pharmacologic treatment", result: "Result (1)", note: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum." },
-  { date: "21/12/2019", type: "Pharmacologic treatment", result: "Result (1)", note: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum." }
-];
-
-const header = ["date", "type", "result"];
+import { CircularProgress } from "@material-ui/core";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import { connect, useSelector } from "react-redux";
+import { loadSummaryData } from "../../../../state/summary/actions";
+import { ISummaryState, SummaryData } from "../../../../state/summary/types";
+import { IState } from "../../../../types";
+import Table from "../../table/Table";
+import { ORDER_BY_DATE_PAGE_SIZE } from "../consts";
+import { IDispatchProps, IStateProps, TProps } from "./../types";
+
+const header = ["date", "type"];
 const order = ["date"];
 const label = {
-  "date": "Date",
-  "type": "Typology",
-  "result": "Result",
-  "note": "Additional notes"
-}
+  date: "Date",
+  type: "Typology",
+};
+
+const PatientSummaryByDate: FunctionComponent<TProps> = ({
+  hasSucceeded,
+  loadSummaryData,
+  isLoading,
+}) => {
+  const [summaryData, setSummaryData] = useState(Array<SummaryData>());
+  const summary = useSelector<IState, ISummaryState>((state) => state.summary);
+  const patient = useSelector((state: IState) => state.patients);
+
+  useEffect(() => {
+    loadSummaryData(patient.selectedPatient?.data?.code!);
+  }, [patient]);
 
-const PatientSummaryByDate: FunctionComponent = () => {
+  useEffect(() => {
+    if (hasSucceeded) {
+      setSummaryData(summary.loadSummaryData.data!);
+    }
+  }, [summary]);
```

Comment:
It seems that this useState (I mean, the one in line 23) is unnecessary because we can handle all this logic regarding summaryData inside redux boundaries. This component, then, would see only a useSelector that grabs the portion of the store that we need.

Reply:
I replaced it.